### DISCUSSION TITLE
Retail eCommerce Funnel Analysis Demo with 1 million members and 87 million record dataset using StarRocks | Demo of StarRocks using Hudi, Iceberg, Delta Lake External Catalog on MinIO + HMS + Apache xTable

### DISCUSSION THREAD

**Discussion by alberttwong** [apache/incubator-xtable#355] [Show and tell]
![Screenshot 2024-02-28 at 9 19 11 AM](https://github.com/apache/incubator-xtable/assets/749093/e77e854f-ef6c-4d7f-9e0c-c9b901eb36c7)

In this tutorial, I'll guide you through working with an 87 million record e-commerce dataset. The data is initially stored in Hudi format and resides on MinIO S3 storage. We'll then use StarRocks to perform queries involving JOINs. Additionally, the tutorial covers creating mirrored datasets in Iceberg and Delta Lake formats using the xtable library.

See more at https://github.com/StarRocks/demo/tree/master/documentation-samples/datalakehouse

**Comment by kywe665** [comment]
Awesome Albert! If you want, you are also welcome to contribute this to the OneTable/XTable blog page: https://onetable.dev/blog/

Documentation how to: https://github.com/apache/incubator-xtable/tree/main/website#blogs

**Reply by alberttwong** [reply]
Done. https://github.com/apache/incubator-xtable/pull/360

**Comment by alberttwong** [comment]
I just wanted to give everyone a FYI that the latest version of Docker Desktop breaks the ecommerce xtable demo at https://xtable.apache.org/blog/ecommerce-funnel-analysis-with-starrocks.

I was doing a demo of it earlier this week and didn't understand why it didn't work until I read https://github.com/docker/for-mac/issues/7286. The recommended fix is to downgrade.
